bypros/proxy.go

```diff
@@ -40,15 +43,22 @@ func (s *Service) Follow(req *Follow) (*EmptyReply, error) {
 	}
 
 	if !s.scID.Equal(req.ScID) {
-		return nil, xerrors.Errorf("wrong skipchain ID: expected '%x', got '%x'", s.scID, req.ScID)
+		s.follow <- struct{}{}
```

Comment:
I don't quite get why we're sending a message on this channel in case of an error? Is it to unblock `s.Follow()` in case of a retry in `listenBlocks`?

Reply:
It is to allow for a new call to `s.Follow()` by the client.

Reply:
Ah yes of course!

Reply:
just to add more precision: this is actually a bug fix, not directly related to the reconnect stuff.